idaes/core/dmf/commands.py

```diff
@@ -391,7 +391,9 @@ def cat_resources(path, objects=(), color=True):
     # get all resources,
     # display any that match an object as a prefix
     for r in d.find():
-        for oid in unmatched:
+        # iterate on a copy of the set to avoid
+        # mutating an object being iterated on
+        for oid in set(unmatched):
```

Comment:
I would make this a `list` and not a `set`: list is faster, doesn't require hashability, and most importantly, is deterministic.  I think I would have gone with:
```python
unmached = list(objects)
for r in d.find():
    for i, oid in enumerate(unmached):
        if not oid:
            continue
        if r.uuid.startswith(oid):
            unmached[i] = None
            # ...
```

Reply:
As a note for posterity (?): I've moved the Pylint-related changes (only incidentally due to Python 3.12), including the one this comment refers to, to their own PR (#1351, now merged as 57cf0a7f31f3b5aafabfe103c5af8d028bb69b37)